sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1360,10 +1382,9 @@ && getLowerBound() instanceof Comparable<?>) {
             ((Comparable<PartitionColumnT>) getLowerBound()).compareTo(getUpperBound()) < EQUAL,
             "The lower bound of partitioning column is larger or equal than the upper bound");
       }
-      checkNotNull(
-          JdbcUtil.JdbcReadWithPartitionsHelper.getPartitionsHelper(getPartitionColumnType()),
-          "readWithPartitions only supports the following types: %s",
-          JdbcUtil.PRESET_HELPERS.keySet());
+      JdbcReadWithPartitionsHelper<PartitionColumnT> helper =
+          MoreObjects.firstNonNull(
+              getPartitionHelper(), JdbcUtil.getDefaultPartitionsHelper(getPartitionColumnType()));
```

Comment:
Are we losing the `"readWithPartitions only supports the following types: %s"` message if there is no custom helper and type is not supported?

Reply:
Also, I think `partitionColumnType` is not a required parameter if custom `partitionsHelper` is passed. Maybe implement something like [this](https://github.com/apache/beam/pull/31733/files#diff-2b0f46ad7242c9feb208ec2a2763286cc39159714a61dccb7fa8f9fa8a48592fL375)?